obspy/clients/filesystem/tsindex.py

```diff
@@ -1550,16 +1357,14 @@ def _fetch_index_rows(self, query_rows=None, bulk_params=None):
                     sa.literal(b).label("station"),
                     sa.literal(c).label("location"),
                     sa.literal(d).label("channel"),
-                    sa.case(
-                        (sa.literal(e) == '*',
-                            sa.literal('0000-00-00T00:00:00')),
-                        (sa.literal(e) != '*', sa.literal(e))
-                        ).label("starttime"),
-                    sa.case(
-                        (sa.literal(f) == '*',
-                            sa.literal('5000-00-00T00:00:00')),
-                        (sa.literal(f) != '*', sa.literal(f))
-                        ).label("endtime")
+                    sa.case((sa.literal(e) == '*',
+                             sa.literal('0000-00-00T00:00:00')),
+                            else_=sa.literal(e)
+                            ).label("starttime"),
+                    sa.case((sa.literal(f) == '*',
+                             sa.literal('5000-00-00T00:00:00')),
+                            else_=sa.literal(f)
+                            ).label("endtime")
```

Comment:
This part is different on the `maintenance_1.4.x` branch, so not sure what needs to be done exactly if it should go into 1.4.1